### ISSUE TITLE
Pelican allows uploads to overwrite already existing objects in the origin outside of Sync

### ISSUE THREAD

**Issue by turetske** [PelicanPlatform/pelican#1695]
Except for rerunning a failed sync, Pelican object put shouldn't allow for the overwriting of already existing files. Objects are supposed to be immutable. We should actually make them so.

This will provide another level of protection against unknowing users who update/alter objects in the origin which can cause an origin/cache mismatch.

**Comment by patrickbrophy** [comment]
Yes this looks like it it does cover the same scope as the issues/PRs mentioned above.

**Comment by brianaydemir** [comment]
> Yes this looks like it it does cover the same scope as the issues/PRs mentioned above.

@patrickbrophy So, can we close this issue, then?
